### PR TITLE
Added pagination for synapse connector

### PR DESCRIPTION
**Pull request by VenkatasivareddyTR**
*Issue #, if available:*
Synapse Connector pagination is not implemented.

*Description of changes:*
Added pagination for synapse connector

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-synapse/src/main/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandler.java [hunk 1]
```diff
@@ -310,6 +318,41 @@ private String encodeContinuationToken(int partition)
...
+
+        int t = token != null ? Integer.parseInt(token) : 0;
+
+        String adjustedSchemaName = caseResolver.getAdjustedSchemaNameString(connection, listTablesRequest.getSchemaName(), configOptions);
+
+        LOGGER.info("Starting pagination at {} with page size {}", t, pageSize);
+        List<TableName> paginatedTables = getPaginatedTables(connection, adjustedSchemaName, t, pageSize);
+        LOGGER.info("{} tables returned. Next token is {}", paginatedTables.size(), t + pageSize);
+
+        String nextToken = paginatedTables.isEmpty() || paginatedTables.size() < pageSize ? null : Integer.toString(t + pageSize);
```
Comment (samarsar) [inline athena-synapse/src/main/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandler.java:336]: There's a chance to get into `listPaginatedTables` method when startToken is not null, but pageSize is UNLIMITED_PAGE_SIZE_VALUE (-1).

With that being said, in the case where start token is 2 and pageSize is -1, this condition will return the incorrect next token. 

paginatedTables.isEmpty() = false
paginatedTables.size() < -1 = false

Then nextToken would be (2 + -1) which would be incorrect.

File: athena-synapse/src/test/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandlerTest.java [hunk 2]
```diff
@@ -342,6 +341,58 @@ public void doGetSplitsContinuation()
...
+        resultSet = mockResultSet(schema, values, new AtomicInteger(-1));
+        Mockito.when(preparedStatement.executeQuery()).thenReturn(resultSet);
+        listTablesResponse = this.synapseMetadataHandler.doListTables(
+                blockAllocator, new ListTablesRequest(this.federatedIdentity, "testQueryId",
+                        "testCatalog", "testSchema", "1", 1));
+        Assert.assertEquals("2", listTablesResponse.getNextToken());
+        Assert.assertArrayEquals(expected, listTablesResponse.getTables().toArray());
+
+
+        // Test 3: Testing single table returned when requesting pageSize 2 signifying end of pagination where nextToken is null.
```
Comment (samarsar) [inline athena-synapse/src/test/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandlerTest.java:380]: Can we add a fourth test where pageSize = UNLIMITED_PAGE_SIZE_VALUE and startToken is not null?

File: athena-synapse/src/main/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandler.java (hunk 1 above)
Comment (samarsar) [reply athena-synapse/src/main/java/com/amazonaws/athena/connectors/synapse/SynapseMetadataHandler.java:336]: This mistake is also in my snowflake pagination review. Will get that fixed ASAP.

**Review by samarsar — CHANGES_REQUESTED** [review]

**Comment by VenkatasivareddyTR** [comment]
Following our Slack discussion, we've decided to skip implementing true pagination for Synapse due to our use of ROW_NUMBER() instead of offset. Therefore, I'm closing this PR.
